### PR TITLE
refactor(console): refactor table component

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

Extract table-related logic into a `<Table />` component.

## Testing

Test locally.
image

### CODE REVIEW

File: packages/console/src/scss/normalized.scss [hunk 1]
```diff
@@ -33,24 +33,14 @@ table {
...
     td {
       font: var(--font-body-medium);
       border-bottom: 1px solid var(--color-divider);
-      padding: _.unit(3) _.unit(2);
-
-      &:first-child,
-      &:last-child {
-        padding: _.unit(3) _.unit(8);
-      }
+      padding: _.unit(3);
```
Comment (charIeszhao) [inline packages/console/src/scss/normalized.scss:43]: Put all table related css styles in the new Table component stylesheet

Comment (xiaoyijun) [reply packages/console/src/scss/normalized.scss:43]: Will do it in later PRs, for we don't replace all tables in this PR.

File: packages/console/src/pages/Applications/index.tsx [hunk 2]
```diff
@@ -89,69 +85,50 @@ const Applications = () => {
...
-              <Button
-                title="applications.create"
-                type="outline"
-                onClick={() => {
-                  navigate({
-                    pathname: createApplicationPathname,
-                    search,
-                  });
-                }}
+        data={applications}
```
Comment (charIeszhao) [inline packages/console/src/pages/Applications/index.tsx:90]: How about calling it `rows` or `rowData`?

File: packages/console/src/pages/Applications/index.tsx [hunk 3]
```diff
@@ -89,69 +85,50 @@ const Applications = () => {
...
+            ),
+          },
+          {
+            title: t('applications.app_id'),
+            dataIndex: 'id',
+            colSpan: 10,
+            render: ({ id }) => <CopyToClipboard value={id} variant="text" />,
+          },
+        ]}
+        emptyContent={
```
Comment (charIeszhao) [inline packages/console/src/pages/Applications/index.tsx:115]: nit: `emptyContent` -> `placeholder`?

File: packages/console/src/scss/normalized.scss (hunk 1 above)
Comment (charIeszhao) [reply packages/console/src/scss/normalized.scss:43]: Ah, I see. OK then, as long as it is planned

File: packages/console/src/pages/Applications/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/Applications/index.tsx:90]: Aha, good idea, let me try

File: packages/console/src/components/Table/types.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import type { FieldPath, FieldPathValue, FieldValues } from 'react-hook-form';
+
+export type Column<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>
+> = {
+  title: ReactNode;
+  dataIndex: TName;
+  render: (value: FieldPathValue<TFieldValues, TName>, row: TFieldValues) => ReactNode;
+  colSpan: number;
```
Comment (charIeszhao) [inline packages/console/src/components/Table/types.ts:11]: colSpan should be optional, and having a default value of 1

File: packages/console/src/pages/Applications/index.tsx [hunk 5]
```diff
@@ -89,69 +85,50 @@ const Applications = () => {
...
-                }}
+        rowGroups={[{ key: 'applications', data: applications }]}
+        rowIndexKey="id"
+        isLoading={isLoading}
+        errorMessage={error?.body?.message ?? error?.message}
+        columns={[
+          {
+            title: t('applications.application_name'),
+            dataIndex: 'name',
+            colSpan: 6,
```
Comment (charIeszhao) [inline packages/console/src/pages/Applications/index.tsx:98]: Why is there a `colSpan: 6`?

Comment (xiaoyijun) [reply packages/console/src/pages/Applications/index.tsx:98]: According to the design 
image

File: packages/console/src/components/Table/types.ts (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/components/Table/types.ts]: updated

File: packages/console/src/pages/Applications/index.tsx (hunk 5 above)
Comment (charIeszhao) [reply packages/console/src/pages/Applications/index.tsx:98]: ok

**Review by charIeszhao — APPROVED** [review]

**Pull request merged**